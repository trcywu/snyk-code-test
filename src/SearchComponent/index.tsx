import React, { FC, useState } from "react";
import { TextField, Button, Container, Box } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles({
  heading: {
    paddingBottom: 16
  },
  textField: {
    paddingRight: 8
  }
});

interface Props {
  onSearch: (name: string) => void;
}
const SearchComponent: FC<Props> = ({ onSearch }) => {
  const classes = useStyles();
  const [name, setName] = useState<string>("");

  const handleChangeForm = (event: React.ChangeEvent<HTMLInputElement>) => {
    setName(event.target.value);
  };

  return (
    <Container>
      <Box className={classes.heading}>Enter your package name here</Box>
      <Box>
        <form
          onSubmit={(e: React.FormEvent<HTMLFormElement>) => {
            e.preventDefault();
            onSearch(name);
          }}
        >
          <TextField
            id="outlined-basic"
            size="small"
            variant="outlined"
            name="name"
            value={name}
            onChange={handleChangeForm}
            className={classes.textField}
          />
          <Button variant="contained" type="submit">
            Search
          </Button>
        </form>
      </Box>
    </Container>
  );
};

export default SearchComponent;
