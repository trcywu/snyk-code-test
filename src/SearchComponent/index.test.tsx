import React from "react";
import renderer from "react-test-renderer";
import SearchComponent from ".";

describe("<SearchComponent />", () => {
  const onSearch = jest.fn();
  it("renders SearchComponent", () => {
    const wrapper = renderer.create(<SearchComponent onSearch={onSearch} />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
