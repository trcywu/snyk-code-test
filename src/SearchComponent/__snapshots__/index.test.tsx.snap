// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchComponent /> renders SearchComponent 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div
    className="MuiBox-root MuiBox-root-12 makeStyles-heading-1"
  >
    Enter your package name here
  </div>
  <div
    className="MuiBox-root MuiBox-root-13"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root MuiTextField-root makeStyles-textField-2"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            disabled={false}
            id="outlined-basic"
            name="name"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-52 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-53"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Search
        </span>
      </button>
    </form>
  </div>
</div>
`;
